apps/chatter/consumers.py

```diff
@@ -1,6 +1,22 @@
 # chat/consumers.py
 import json
 from channels.generic.websocket import AsyncWebsocketConsumer
+from rest_framework.authtoken.models import Token
+from channels.db import database_sync_to_async
+from .models import Message
+from .models import Room
+
+@database_sync_to_async
+def get_user(token_key):
+    try:
+        token = Token.objects.get(key=token_key)
+        return token.user
+    except Token.DoesNotExist:
+        return AnonymousUser()
```

Comment:
It is used in the same file. The reason it is broken into its own function is the @database_sync_to_async wrapper without which it complains that it can't be called from an async context.

Reply:
Ah, I forgot Github show's more than 1 line.. I was referring to line 15: `AnonymousUser`. Does not appear to be defined or imported?

Reply:
Oh woops, gotcha. It's imported now.